analysis/__main__.py

```diff
@@ -0,0 +1,43 @@
+import glob
+import sys
+
+import pandas as pd
+
+from heuristic.classes import Problem, Solution
+from .statistics import STATISTICS
+
+
+def main():
+    """
+    Prints several statistics of each instance and descriptive
+    statistics of the instances overall to a file.
+    """
+    if len(sys.argv) < 2:
+        raise ValueError(f"{sys.argv[0]}: expected file location.")
+
+    cumulative_df = pd.DataFrame()
+
+    for instance in glob.glob(sys.argv[1]):
+        problem = Problem.from_file(instance, delimiter=',')
+        solution = Solution.from_file(f"solutions/oracs_{problem.instance}.csv")
+        instance_df = pd.DataFrame()
+
+        for statistic in STATISTICS:
+            statistic_value = statistic(solution)
+            instance_df[statistic_value[0]] = [statistic_value[1]]
+
+        cumulative_df = pd.concat([cumulative_df, instance_df])
+
+        problem.clear()
```

Comment:
I think this will need to be `Problem.clear()` once you've moved clear into `Singleton`.